### ISSUE TITLE
type changed. typeof Model -> typeof ModelType

### ISSUE THREAD

**Issue by sdg9670** [sequelize/sequelize-auto#511]
refrenced issue:
https://github.com/sequelize/sequelize/pull/13010

typeof Model changed to typeof ModelType since 6.6.1 verseion.

![image](https://user-images.githubusercontent.com/40787013/112413936-7b4fb780-8d64-11eb-93ea-7d90aafdebb6.png)

**Comment by steveschmitt** [comment]
Ah, thanks.  I'll have to update the generator.

**Comment by KapitanOczywisty** [comment]
@steveschmitt casting shouldn't be needed anymore.

**Comment by papb** [comment]
Actually it changed from `typeof Model` to `ModelType`. I did not expect this change to break anything though. I'm very sorry if that's the case.

@KapitanOczywisty can you clarify what you mean?

**Comment by KapitanOczywisty** [comment]
@papb  From what I can see in the code, `as typeof Model` was added to "fix" issue which PR is solving in the first place.

> Actually it changed from `typeof Model` to `ModelType`.

To be precise, `ModelType` changed, but wasn't used before. TypeScript doesn't really care about the name in the most cases, only about real signature. Through `typeof Mode` IS still assignable to the `ModelType` in typescript <= 4.2. Mentioned problem starts at ts 4.3, where abstractness is checked as well, at least at the time of writing, because it's still in the development.

You can fix this easily by adding `abstract` to the constructor definition ([TypeScript 4.3 Playground](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210328#code/PTAEEMDsE9QMwPYCdQBcCmBnVmCwAoAYwUmwhgGESyAuc2AXlEgFcAbNgbgIJFGQCWAcwGRwbUMQAm6HvnAAjbEnCFUktuEyZQAWQQy2AHgAqAGlABVAHygA3gF856AB4AHZOtTQ36PQfQ2Ex90U31DAEFUVCQBBRYMHSYoaAsTCiR0cFQBEiiYuISsUCYTcMD82PjE2yZIdAB3UAAKAEoS23LjMoC2SsLEtIysnLzoqqLMa258AjgWSDVcyDQsVGaAW166LuDfVsc5RPWUqlJ1LTQQhDh-Q1aZ3jA4ARd0KTlFZVV1Qk1tO6BABMpgsNnsTlm+FcHiQXhCgKCIRBPUi4wGxWSMCGmWyy361UxoFRFXRhKmJQgShiP2YjRa7QYnV6YV6BMmOJG+LJk2mcnmi1GK2OQM220Re3QQIOkIIIuap2oFx03l8N0R0pmQA)). The same should be done in the RobinBuschmann/sequelize-typescript
```ts
export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = abstract new () => Model<TModelAttributes, TCreationAttributes>;
```

**Comment by KapitanOczywisty** [comment]
_I've rewritten this answer, since I've messed up some facts first time_

@papb  From what I can see in the code, `as typeof Model` was added to "fix" issue which PR is solving in the first place.

> Actually it changed from `typeof Model` to `ModelType`.

To be precise, `ModelType` changed, and should be used before anyway. `as typeof Model` shouldn't be used, it would be compatible if we would add `abstract` to the definition, but this is only possible since TypeScript 4.2 thus typings wouldn't be backward compatible ([TypeScript Playground](https://www.typescriptlang.org/play?ts=4.3.0-dev.20210328#code/PTAEEMDsE9QMwPYCdQBcCmBnVmCwAoAYwUmwhgGESyAuc2AXlEgFcAbNgbgIJFGQCWAcwGRwbUMQAm6HvnAAjbEnCFUktuEyZQAWQQy2AHgAqAGlABVAHygA3gF856AB4AHZOtTQ36PQfQ2Ex90U31DAEFUVCQBBRYMHSYoaAsTCiR0cFQBEiiYuISsUCYTcMD82PjE2yZIdAB3UAAKAEoS23LjMoC2SsLEtIysnLzoqqLMa258AjgWSDVcyDQsVGaAW166LuDfVsc5RPWUqlJ1LTQQhDh-Q1aZ3jA4ARd0KWd3TyvfO8C99AAJjCvX61WKyRgQ0y2WWYMmJVAPUi4wGWFqECUMVU6nqTTaHT+3S68MGSOGsLGBXBU0e+Hmi1GK2OgM22yJAMBBycs3wLOap2oFx03l8NyJDwIQA)).
```ts
export type ModelType<TModelAttributes = any, TCreationAttributes = TModelAttributes> = abstract new () => Model<TModelAttributes, TCreationAttributes>;
```

However, as mentioned above It's not needed anymore, user Models, even with attributes, are now accepted without any errors.
